lib/super_digit.rb

```diff
@@ -1,12 +1,54 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+
+# Given an integer, we need to find the super digit of the integer.
+
+# If the number has only digit, then its super digit is that number. Otherwise, the super digit of x is equal to the super digit of the sum of the digits of x. For example, the super digit of 9875 will be calculated as:
+
+# super_digit(9875) --> superdigit(9 + 8 + 7 + 5) = superdigit(29)
+# super_digit(29)   --> superdigit(2 + 9) = superdigit(11)
+# super_digit(11)   --> superdigit(1 + 1) = superdigit(2)
+# super_digit(2) --> 2
+
+
+# Time Complexity - O(n)
+# Space Complexity - O(n)
 def super_digit(n)
```

Comment:
The loop you have here and the map methods are O(n), then you begin the recursion.  So this is going to be more than O(n)